cmds/crc/crc.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Prints crc checksum of a file.
+//
+// Synopsis:
+//     crc OPTIONS [FILE]
+//
+// Description:
+//     One of the crc types must be specified. If there is no file, stdin is
+//     read.
+//
+// Options:
+//     -crc32: Use CRC-32 IEEE standard
+//     -crc64: Use CRC-64 ECMA standard
+package main
+
+import (
+	"flag"
+	"fmt"
+	"hash/crc32"
+	"hash/crc64"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+var (
+	crc32IEEE = flag.Bool("crc32", false, "CRC-32 IEEE")
+	crc64ECMA = flag.Bool("crc64", false, "CRC-64 ECMA")
+)
+
+func main() {
+	flag.Parse()
+	if *crc32IEEE && *crc64ECMA {
+		log.Fatal("cannot specify more than one crc type")
+	}
+
+	var r io.Reader
+	switch flag.NArg() {
+	case 0:
+		r = os.Stdin
+	case 1:
+		f, err := os.Open(flag.Arg(0))
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer f.Close()
+		r = f
+	default:
+		log.Fatal("expected 0 or 1 positional args")
+	}
+	data, err := ioutil.ReadAll(r)
```

Comment:
I think you should make this work in a streaming mode. The package supports this, see
https://golang.org/src/hash/crc32/crc32.go?s=6349:6401#L200

Reply:
Done!